ignite/handlers/checkpoint.py

```diff
@@ -269,7 +272,10 @@ def __call__(self, engine: Engine) -> None:
             if not isinstance(priority, numbers.Number):
                 raise ValueError("Output of score_function should be a number")
         else:
-            priority = engine.state.get_event_attrib_value(Events.ITERATION_COMPLETED)
+            if self.global_step_transform is not None:
+                priority = global_step
```

Comment:
It looks generally good, but I have a minor correction to make here. It is not obvious that 
`global_step` is defined here.
It would be better to code it like that :
```python
        global_step = None
        if self.global_step_transform is not None:
            global_step = self.global_step_transform(engine, engine.last_event_name)
            suffix = "{}".format(global_step)
       # ....
        else:
            if global_step is not None:
                priority = global_step
            else:
                priority = engine.state.get_event_attrib_value(Events.ITERATION_COMPLETED)
```

What do you think ?

Reply:
Yes, I think that's more clear. Changed.